### ISSUE TITLE
Using EF.Functions.Count without arguments (`COUNT(*)`)

### ISSUE THREAD

**Issue by InspiringCode** [zompinc/efcore-extensions#18]
Is there any way I could write a query that results in `SELECT X, COUNT(*) OVER() TotalCount FROM y`?

I tried `EF.Functions(1, EF.Functions.Over())` and `EF.Functions(new { One = 1 }, EF.Functions.Over())` and a few other variants but they all throw an exception. Would it be possible to provide an overload of `Count` that takes no query parameters and just produces `COUNT(*)` in this case?

**Comment by virzak** [comment]
Yeah, should be able to do that soon. In the meantime you can insert any non nullable column. If all your columns are nullable you can do this;

```cs
var query = DbContext.TestRows
.Select(r => EF.Functions.Count<int, TResult>(r.Col1 == null ? 1 : 1, EF.Functions.Over()));
```

**Comment by virzak** [comment]
@InspiringCode, this is available starting version [8.0.18-beta](https://www.nuget.org/packages/Zomp.EFCore.BinaryFunctions.SqlServer/8.0.18-beta)
